log-collector/src/main/scala/io/phdata/pulse/logcollector/LogCollectorRoutes.scala

```diff
@@ -79,10 +81,27 @@ class LogCollectorRoutes(solrService: SolrService) extends JsonSupport with Lazy
       // create a streaming Source from the incoming json
       entity(as[Array[LogEvent]]) { logEvents =>
         logger.trace("received message")
-        logEvents.foreach(logEvent => streamRef ! (applicationName, logEvent))
+        logEvents.foreach(logEvent =>
+          streamRef ! (applicationName, Util.logEventToFlattenedMap(logEvent)))
+
+        complete(HttpEntity(ContentTypes.`application/json`, "ok"))
+      }
+    }
+  } ~ path("v1" / "json" / Segment) { applicationName =>
+    /**
+     * Consumes an single json event
+     */
+    post {
+      // create a streaming Source from the incoming json string
+      entity(as[String]) { logEvent =>
```

Comment:
Can you change this to accept an array of events by default? That will always be more efficient. We need to deprecate the endpoint that accepts a single log event.

Reply:
Would this not cause a problem when someone sends something that doesn't conform to LogEvent like `{my_metric: 20.2, timestamp: 2019-03-06T01:14:11}`.  Or are you saying it should take `Array[String]` ?

Reply:
A user should be able to send a JsonArray[JsonObject], you can still do your casting to `as[String]`, the change is to parse it into an array of objects and `foreach` them onto the stream